src/v/kafka/server/group_manager.cc

```diff
@@ -214,50 +214,62 @@ ss::future<> group_manager::handle_partition_leader_change(
      * partition.
      */
     p->loading = true;
```

Comment:
**Can you state in the commit message why p->loading isn't sufficient?**

Updated commit message:

> We already have the p->loading flag to prevent processing new requests once the reload started but this mechanism doesn't drain the ongoing requests so there is a chance for the reload process to interleave with the old request processing which may lead to the incorrect state

**Is p->loading still relevant or can this rwlock replace it?**

I need to think about it but it'll definitely will increase the scope let's figure it out later.

**... but this commit only adds a lock around leader change**

The follow up commits add transactional method: begin_tx, prepare_tx, commit_tx, abort_tx, txn_offset_commit